Detectors/Raw/test/testRawReaderWriter.cxx

```diff
@@ -82,23 +89,37 @@ struct SimpleRawWriter { // simple class to create detector payload for multiple
 
     // create payload for every interaction and push it to writer
     for (const auto& ir : irs) {
-      for (int icru = 0; icru < NCRU; icru++) {
+      for (int icru = 0; icru < NCRU - 1; icru++) {
         // we will create non-0 payload for all but 1st link of every CRU, the writer should take care
         // of creating empty HBFs for the links w/o data
         for (int il = 0; il < NLinkPerCRU; il++) {
+          buffer.clear();
           int nGBT = gRandom->Poisson(HBFUtils::MAXCRUPage / HBFUtils::GBTWord * (il));
           if (nGBT) {
-            buffer.resize((nGBT + 2) * HBFUtils::GBTWord); // reserve 16B words accounting for the Header and Trailer
+            buffer.resize((nGBT + 2) * HBFUtils::GBTWord, icru * NLinkPerCRU + il); // reserve 16B words accounting for the Header and Trailer
             std::memcpy(buffer.data(), PLHeader.c_str(), HBFUtils::GBTWord);
             std::memcpy(buffer.data() + buffer.size() - HBFUtils::GBTWord, PLTrailer.c_str(), HBFUtils::GBTWord);
             // we don't care here about the content of the payload, except the presence of header and trailer
-          } else {
-            buffer.clear();
           }
           writer.addData((icru << 8) + il, icru, il, 0, ir, buffer);
         }
       }
     }
+    // fill special CRU with preformatted pages
```

Comment:
@davidrohr this block shows how the fixed pages can be added to the link buffer.